index.src.html

```diff
@@ -3192,40 +3283,56 @@
-          |displayHeight| to {{VideoFrame/[[coded width]]}},
-          {{VideoFrame/[[coded height]]}}, {{VideoFrame/[[visible left]]}},
-          {{VideoFrame/[[visible top]]}}, {{VideoFrame/[[visible width]]}},
-          and {{VideoFrame/[[visible height]]}} respectively.
-      10. Assign |duration| and |timestamp| to {{VideoFrame/[[duration]]}} and
-          {{VideoFrame/[[timestamp]]}} respectively.
-  2. Return |frame|.
+: <dfn>Create a VideoFrame</dfn> (with |output|, |timestamp|, |duration|, |displayAspectWidth|, |displayAspectHeight|, and |colorSpace|)
+:: 1. Let |frame| be a new {{VideoFrame}}, constructed as follows:
+        1. Assign `false` to {{platform object/[[Detached]]}}.
+        2. Let |resource| be the [=media resource=] described by |output|.
+        3. Let |resourceReference| be a reference to |resource|.
+        4. Assign |resourceReference| to {{VideoFrame/[[resource reference]]}}.
+        5. If |output| uses a recognized {{VideoPixelFormat}}, assign that format to
+            {{VideoFrame/[[format]]}}. Otherwise, assign `null` to
+            {{VideoFrame/[[format]]}}.
+        6. Let |codedWidth| and |codedHeight| be the coded width and height of the
+            |output| in pixels.
+        8. Let |visibleLeft|, |visibleTop|, |visibleWidth|, and |visibleHeight| be
+            the left, top, width and height for the visible rectangle of |output|.
+        7. Let |displayWidth| and |displayHeight| be the the display size of
+            |output| in pixels.
+        8. If |displayAspectWidth| and |displayAspectHeight| are provided,
+            increase |displayWidth| or |displayHeight| until the ratio of
+            |displayWidth| to |displayHeight| matches the ratio of
+            |displayAspectWidth| to |displayAspectHeight|.
+        9. Assign |codedWidth|, |codedHeight|, |visibleLeft|, |visibleTop|,
+            |visibleWidth|, |visibleHeight|, |displayWidth|, and
+            |displayHeight| to {{VideoFrame/[[coded width]]}},
+            {{VideoFrame/[[coded height]]}}, {{VideoFrame/[[visible left]]}},
+            {{VideoFrame/[[visible top]]}}, {{VideoFrame/[[visible width]]}},
+            and {{VideoFrame/[[visible height]]}} respectively.
+        10. Assign |duration| and |timestamp| to {{VideoFrame/[[duration]]}} and
+            {{VideoFrame/[[timestamp]]}} respectively.
+        11. Assign {{VideoFrame/[[color space]]}} with the result of running the
+            [=VideoFrame/Pick Color Space=] algorithm, with |colorSpace| and
+            {{VideoFrame/[[format]]}}.
+    2. Return |frame|.
+
+
+: <dfn for=VideoFrame>Pick Color Space</dfn> (with |overrideColorSpace| and |format|)
+:: 1. If |overrideColorSpace| is provided, return a new {{VideoColorSpace}}
+        constructed with |overrideColorSpace|.
+
+        NOTE: User Agents may replace `null` members of the provided
+            |overrideColorSpace| with  guessed values as determined by
+            implementer defined hueristics.
```

Comment:
`s/hueristics/heuristics/`.

That smells like a compatibility problem, can we not spell out some rules here? Isn't this likely to be a simple extrapolation from a field to the obvious others, as defined around line 181 ?

Reply:
In most cases these should be obvious extrapolations, yes. I'm not sure I have the expertise to prescribe a process though, and limiting the implementation to changing empty fields allows apps to be specific and avoid implementation-defined behavior if desired.